Anguandia/quickcredit
# Add feature to modify loan in database

#### Purpose of this pull request

Enable update of loan entry in database

#### Tasks to be completed

- Add availability check middleware in update route to run after user input validation
- Add query to read confirmed target loan from database
- Reinstantiate read loan to utilize instance methods approve and set
payment installation. Approve auto disburses total loan value to account
(credits loan) and updates current balance
- Add id setter method in loan model to accommodate re instantiation from database object
- Add a second query to update the target's status in the database

#### How to manually test this feature

- Clone this repo and run npm install to install this app locally
- Start the app with either npm start or debug=quickcredit npm run dev commands in your terminal
- Proceed to create a loan with a POST/api/v1/loans request supplying a json object with values for the fields of email(user), amount(float) and tenor(integer)
- Approve or reject the loan with a PATCH/api/v1/loans/<loanId> request with a body the json object with key status and value rejected or approved
- The response will show the loan with the new status
- Try changing the casing of the status value, the system should be insensitive to case
- Try using a status value other than rejected or approved, the response should reject your status value
- Try any conceivable user errors and observe the response

[finishes]
